app/controllers/users_controller.rb

```diff
@@ -19,30 +19,31 @@ def user_guardians
     if admin?
       @guardians = current_user.guardians.all
       raise ActiveRecord::RecordNotFound unless @guardians
+
       render json: @guardians, status: :ok
     else
       render json: { error: 'Not Allowed' }, status: :unauthorized
     end
   end
 
   def add_guardian
-    if admin?
-      if User.find_by_email(user_params[:email])
-        render json: { error: 'Email Exist , try a diffrent one' }, status: :not_acceptable
+    return unless admin?
+
+    if User.find_by_email(user_params[:email])
+      render json: { error: 'Email Exist , try a diffrent one' }, status: :not_acceptable
+    else
+      @guardian = current_user.guardians.create(user_params)
```

Comment:
First question: does it mean that quardian takes over all wards from the logged-in user?
I am not sure if I understand the use case :)

But - no matter what - it deserves to be moved to a model :) Let's give it a try!